lib/paths.cf

```diff
@@ -104,7 +104,9 @@ bundle common paths
 
     !(freebsd|darwin|smartos)::
       "path[npm]"      string => "/usr/bin/npm";
-      "path[pip]"      string => "/usr/bin/pip";
+      "path[pip]"      string => ifelse(
+        fileexists("/usr/bin/pip3"), "/usr/bin/pip3",
+        "/usr/bin/pip");
```

Comment:
Is there ever any reason that both pip and pip3 would be present concurrently (it's the case on my system).

Is there ever any reason that someone would actually use both pip and pip3 concurrently? If so, seems more appropriate to introduce pip3.

Reply:
Indeed. So then if they are the same, pip is a symlink to pip3 or if they are different, pip is pip2 and pip3 is pip3, then the legacy pip package promise might need to have an attribute about which pip to use?

As it is, the legacy pip package handler relies on this specific `$(paths.pip)` so that would need to be refactored somewhat to support choosing?

https://github.com/cfengine/masterfiles/blob/master/lib/packages.cf#L363

The actual package support is here and uses `$(paths.pip)` https://github.com/cfengine/masterfiles/blob/master/lib/packages.cf#L407

So I would suppose that we could add a pip3 path, but set `paths.pip` to pip3 if pip3 is available?